### ISSUE TITLE
Macro still crashes with SIGSEGV when non-ASCII character in string template literal with interpolation

### ISSUE THREAD

**Issue by maxmilton** [oven-sh/bun#3830]
### What version of Bun is running?

0.7.0

### What platform is your computer?

Linux 6.4.6-arch1-1 x86_64 unknown

### What steps can reproduce the bug?

1. Create file `macro.ts`:
```ts
export function fn(str) {
  return str;
}
```

2. Create file `index.test.ts`:
```ts
import { expect, test } from 'bun:test';
import { fn } from './macro' assert { type: 'macro' };

test('works', () => {
  expect(fn(`A${''}`)).toBe('A');
});

test('segfaults', () => {
  expect(fn(`©${''}`)).toBe('©');
});
```

3. Run `bun test`.

### What is the expected behavior?

Template literals are processed correctly as Unicode when passed as macro arguments.
Bun process runs without a segmentation fault.

### What do you see instead?

```
~/P/_/bun-macro-sigsegv ❱ bun test
bun test v0.7.0 (aa1ad7f0)

index.test.ts:
fish: Job 1, 'bun test' terminated by signal SIGSEGV (Address boundary error)
```

### Additional information

This is a continuation of https://github.com/oven-sh/bun/issues/3641

Now that bun v0.7.0 was released, it looks like the original bug has not been fixed; a bun macro with arg string template literal with non-ASCII characters + interpolation still crash with a segfault.

Same result in both bun stable `v0.7.0 (aa1ad7f0)` and canary `v0.7.1 (f3200ac0)`.

**Comment by Jarred-Sumner** [comment]
This is a slightly different case. For the conversion to happen successfully, minification must be enabled so that the string gets simplified. The crash is happening due to incorrect error handling logic when throwing the error. Haven't narrowed down why yet.

**Comment by maxmilton** [comment]
When experimenting with workarounds, I found this is also an issue with plain strings when there is string concatenation:

```ts
import { expect, test } from 'bun:test';
import { fn } from './macro' assert { type: 'macro' };

test('works 1', () => {
  expect(fn('A')).toBe('A');
});

test('works 2', () => {
  expect(fn('©')).toBe('©');
});

test('works 3', () => {
  expect(fn('A' + '')).toBe('A');
});

test('segfaults', () => {
  expect(fn('©' + '')).toBe('©');
});
```

This aligns with Jarred's note about the string needing to be simplified and still confirms it only happens for non-ASCII characters.

Another weird case that may warrant its own issue, a string with concatenation segfaults when passed in a variable (including ASCII characters):

```ts
import { expect, test } from 'bun:test';
import { fn } from './macro' assert { type: 'macro' };

test('works 1', () => {
  expect(fn('A')).toBe('A');
});

test('works 2', () => {
  const text = 'A';
  expect(fn(text)).toBe('A');
});

test('works 3', () => {
  expect(fn('A' + '')).toBe('A');
});

test('segfaults', () => {
  const text = 'A' + '';
  expect(fn(text)).toBe('A');
});
```

In this case `const text = 'A' + ''` is not statically known without first concatenating the 2 strings, so that's fair enough, but it is a bit of a gotcha since passing `'A' + ''` directly to a macro does work.

Also worth noting that last case is not a problem for numbers:

```ts
...

test('works', () => {
  const num = 1 + 2
  expect(fn(num)).toBe(3);
});
```

**Comment by maxmilton** [comment]
For anyone seeking a workaround, you can use another macro for preprocessing; to manually do string processing like interpolation or concatenation.

`macros.ts`:

```ts
export function interpolate(text: string, values: string[]): string {
  let result = text;

  values.forEach((value, index) => {
    result = result.replace(new RegExp(`%%${index + 1}%%`, 'g'), value);
  });

  return result;
}

export function echo<T>(value: T): T {
  return value;
}
```

`index.ts`

```ts
import { echo, interpolate } from './macros' assert { type: 'macro' };

const result = echo(interpolate('© %%1%% %%2%%', ['val', '😸'])); // © val 😸
```

Needing an extra layer of processing is unintuitive but it does work.

**Comment by Electroid** [comment]
Closed by #6756

**Comment by maxmilton** [comment]
Although it no longer segfaults, the underlying issue is still observable. Instead of a segfault we now see an error:

```
bun test v1.0.8 (53d1acb0)

index.test.ts:


error: "Cannot convert argument type to JS" error in macro

  expect(fn(`©${''}`)).toBe('©');
         ^
/home/max/Projects/_experiments/bun-test-sigsegv2/index.test.ts:9:10 193

 0 pass
 1 fail
Ran 1 tests across 1 files. [8.00ms]
```

It's nice the bun process doesn't crash, however I wouldn't consider the issue resolved. Should we rename the title of this issue to something like `Macro unable to handle non-ASCII character in string template literal with interpolation`, or should I create a new issue?
